crypto/aes_gcm.go

```diff
@@ -0,0 +1,167 @@
+package crypto
+
+import (
+	"crypto/aes"
+	"crypto/cipher"
+	"crypto/rand"
+	"crypto/sha512"
+	"encoding/json"
+	"fmt"
+	"hash"
+
+	"github.com/bndw/pick/errors"
+	"golang.org/x/crypto/pbkdf2"
+)
+
+type AESGCMClient struct {
```

Comment:
Rename sounds good to me.

I don't think there is anything else outstanding on this PR? Once this is in and squashed, let's do a sanity check and final testing, and get this merged in :)